sdk/core/Azure.Core/src/ResponseError.cs

```diff
@@ -18,7 +18,8 @@ namespace Azure
     [TypeReferenceType(true, new string[] { nameof(Target), nameof(Details) })]
     public sealed class ResponseError
     {
-        private readonly JsonElement _element;
+        // This field needs to be internal rather than private so that it can be used by the System.Text.Json source generator
+        internal readonly JsonElement _element;
```

Comment:
This is surprising to me. Doesn't the source generator pick up the `SerializationConstructor`?

Reply:
I'll try without - I had been getting this vague error - `SYSLIB1030: System.Text.Json source generator did not generate output for type ResponseError` and it was a bit of trial and error trying to figure out why.